### PR TITLE
[REF] mail, *: make rtc invitations use the new ChannelMember model

### PR DESCRIPTION
**Pull request by ThanhDodeurOdoo**
* test_discuss_full

channel invitation to rtc calls are now using the `ChannelMember` and
`mail.channel.partner` models instead of the `partner` and `guest`
models.

part of task-2692836

### CODE REVIEW

File: addons/mail/static/src/models/channel_member.js [hunk 1]
```diff
@@ -57,6 +57,10 @@ registerModel({
             readonly: true,
             required: true,
         }),
+        invitingThread: one('Thread', {
+            inverse: 'invitedMembers',
+            isCausal: true,
+        }),
```
Comment (seb-odoo) [inline addons/mail/static/src/models/channel_member.js:63]: field not used an isCausal is wrong here (should not delete thread on deleting member)

File: addons/mail/models/mail_channel.py [hunk 2]
```diff
@@ -345,6 +345,13 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
         partners = self.env['res.partner'].browse(partner_ids or []).exists()
         guests = self.env['mail.guest'].browse(guest_ids or []).exists()
         notifications = []
+        if invite_to_rtc_call:
+            if self.env.user._is_public() and 'guest' in self.env.context:
+                current_guest = self.env.context.get('guest')
```
Comment (seb-odoo) [inline addons/mail/models/mail_channel.py:350]: use `_get_guest_from_context`

File: addons/mail/models/mail_channel.py [hunk 3]
```diff
@@ -345,6 +345,13 @@ def add_members(self, partner_ids=None, guest_ids=None, invite_to_rtc_call=False
         partners = self.env['res.partner'].browse(partner_ids or []).exists()
         guests = self.env['mail.guest'].browse(guest_ids or []).exists()
         notifications = []
+        if invite_to_rtc_call:
```
Comment (seb-odoo) [inline addons/mail/models/mail_channel.py:348]: condition look unrelated to what is done inside

File: addons/mail/models/mail_channel.py [hunk 4]
```diff
@@ -826,8 +817,7 @@ def channel_info(self):
 
             # add RTC sessions info
             info.update({
-                'invitedGuests': [('insert', [{'id': member.guest_id.id, 'name': member.guest_id.name} for member in invited_members_by_channel[channel] if member.guest_id])],
-                'invitedPartners': [('insert', [{'id': member.partner_id.id, 'name': member.partner_id.name} for member in invited_members_by_channel[channel] if member.partner_id])],
+                'channel': [('insert', {'id': channel.id, 'invitedMembers': [('insert', invited_members_by_channel[channel].mail_channel_member_format())]})],
```
Comment (seb-odoo) [inline addons/mail/models/mail_channel.py:820]: `invitedMembers` is still on thread for now, not on channel?

Comment (seb-odoo) [reply addons/mail/models/mail_channel.py:820]: or is that the wrongly named channel that is actually thread?
but if that is the case, `model` should be specified together with `id`

File: addons/mail/models/mail_channel_partner.py [hunk 5]
```diff
@@ -121,14 +121,39 @@ def _get_as_sudo_from_request(self, request, channel_id):
             return guest.env['mail.channel.partner'].sudo().search([('channel_id', '=', channel_id), ('guest_id', '=', guest.id)], limit=1)
         return self.env['mail.channel.partner'].sudo()
 
+    def mail_channel_member_format(self):
+        member_formatted_data = []
```
Comment (seb-odoo) [inline addons/mail/models/mail_channel_partner.py:125]: should be plural

File: addons/mail/models/mail_channel_partner.py [hunk 6]
```diff
@@ -121,14 +121,39 @@ def _get_as_sudo_from_request(self, request, channel_id):
...
+        for record in self:
```
Comment (seb-odoo) [inline addons/mail/models/mail_channel_partner.py:126]: record -> member

File: addons/mail/models/mail_channel_partner.py [hunk 7]
```diff
@@ -170,46 +193,36 @@ def _rtc_sync_sessions(self, check_rtc_session_ids=None):
...
             }))
         self.env['bus.bus']._sendmany(invitation_notifications)
-        if invited_guests or invited_partners:
+        if members:
             channel_data = {'id': self.channel_id.id}
-            if invited_guests:
-                channel_data['invitedGuests'] = [('insert', [{'id': guest.id, 'name': guest.name} for guest in invited_guests])]
-            if invited_partners:
-                channel_data['invitedPartners'] = [('insert', [{'id': partner.id, 'name': partner.name} for partner in invited_partners])]
+            if members:
```
Comment (seb-odoo) [inline addons/mail/models/mail_channel_partner.py:225]: 2 times if members

File: addons/mail/models/mail_channel.py [hunk 8]
```diff
@@ -456,45 +456,35 @@ def _can_invite(self, partner_id):
     # RTC
     # ------------------------------------------------------------
 
-    def _rtc_cancel_invitations(self, partner_ids=None, guest_ids=None):
-        """ Cancels the invitations of the RTC call from all invited members (or the specified partner_ids).
-            :param list partner_ids: list of the partner ids from which the invitation has to be removed
-            :param list guest_ids: list of the guest ids from which the invitation has to be removed
-            if either partner_ids or guest_ids is set, only the specified ids will be invited.
+    def _rtc_cancel_invitations(self, member_ids=None):
+        """ Cancels the invitations of the RTC call from all invited members (or the specified partner_ids),
```
Comment (seb-odoo) [inline addons/mail/models/mail_channel.py:460]: obsolete reference to partner_ids here

File: addons/mail/static/src/models/thread.js [hunk 9]
```diff
@@ -1945,15 +1941,11 @@ registerModel({
...
-         * in which case the two following fields should be a single relation to that model.
-         */
-        invitedGuests: many('Guest'),
-        /**
-         * List of partners that have been invited to the RTC call of this channel.
+         * List of members that have been invited to the RTC call of this channel.
          */
-        invitedPartners: many('Partner'),
+        invitedMembers: many('ChannelMember', {
+            isCausal: true,
```
Comment (seb-odoo) [inline addons/mail/static/src/models/thread.js:1947]: no isCausal here either, member shouldn't be removed from channel when no longer invited

**Review by seb-odoo — APPROVED** [review]
@robodoo r+ rebase-ff

**Comment by ThanhDodeurOdoo** [comment]
@robodoo retry
